backend/src/discovita/api/dependencies.py

```diff
@@ -1,41 +1,40 @@
 """API dependencies."""
 
-from typing import AsyncGenerator
-from fastapi import Depends
-from httpx import AsyncClient, Limits, Timeout
-
 from discovita.config import Settings
 from discovita.dependencies import get_settings
 from discovita.service.coach.prompt.manager import PromptManager
 from discovita.service.coach.service import CoachService
-from discovita.service.openai.client.client import OpenAIClient
-from discovita.service.openai.image_description import ImageDescriptionService
-from discovita.service.openai.image_generation import ImageGenerationService
+from discovita.service.openai.core import OpenAIService
+from discovita.service.openai.core.image_description import ImageDescriptionService
+from discovita.service.openai.core.image_generation import ImageGenerationService
+from fastapi import Depends
+
 
 async def get_openai_client(
-    settings: Settings = Depends(get_settings)
-) -> OpenAIClient:
+    settings: Settings = Depends(get_settings),
+) -> OpenAIService:
     """Get OpenAI client."""
-    return OpenAIClient(
+    return OpenAIService(
         api_key=settings.openai_api_key,
-        base_url="https://api.openai.com/v1",
-        test_mode=False
     )
 
+
 async def get_image_description_service(
-    client: OpenAIClient = Depends(get_openai_client)
+    client: OpenAIService = Depends(get_openai_client),
 ) -> ImageDescriptionService:
     """Get image description service."""
     return ImageDescriptionService(client)
 
+
 async def get_image_generation_service(
-    client: OpenAIClient = Depends(get_openai_client)
+    client: OpenAIService = Depends(get_openai_client),
```

Comment:
Rename variable to` open_ai_service` rather than `client`.  I'd like to be clear in the code what is a service and what is a client.  A service contains business logic and may contain a client inside of it.  A client has bare-minimum business logic, is a thinnest-possible wrapper around a 3pAPI.  Name your variable accordingly.